src/Application/QueryStringParser.php

```diff
@@ -38,11 +43,33 @@ private function splitQueryString( string $queryString ): array {
 		);
 	}
 
+	private function isInstancePart( string $part ): bool {
+		return str_starts_with( $part, 'haswbstatement:' );
+	}
+
 	private function isFacetPart( string $part ): bool {
 		return str_starts_with( $part, 'haswbfacet:' )
 			|| str_starts_with( $part, '-haswbfacet:' );
 	}
 
+	/**
+	 * @param array<string, NumericPropertyId|ItemId> $instance
+	 * @return array<string, NumericPropertyId|ItemId>
+	 */
+	private function handleInstancePart( string $part, array &$instance ): array {
+		$part = substr( $part, strlen( 'haswbstatement:' ) );
+		[ $propertyIdString, $itemIdString ] = explode( '=', $part, 2 );
+
+		if ( $propertyIdString === '' || $itemIdString === '' ) {
+			return [];
+		}
+
+		$instance['propertyId'] = new NumericPropertyId( $propertyIdString );
```

Comment:
@JeroenDeDauw I am not sure what type should whether we should keep `propertyId` as the type or we should just use string, since it is only used by `UiBuilder`.

Reply:
Good to use the PropertyId type, since this thing is being passed around between multiple classes

We probably should put this into a try-catch, because the constructor will throw on invalid property IDs. This should have an accompanying unit test